specs/data-grid/data-grid.spec.md

```diff
@@ -0,0 +1,251 @@
+
+- `columnsData`  
+An array of `DataGridColumn` objects that define what columns will be displayed in the grid.  The order of the columns determines their order in the grid.
+
+- `rowItemTemplate`  
+Custom [template](https://fast.design/docs/fast-element/declaring-templates) to use when generating rows by iterating over data.  The default template uses `fast-data-grid-row`, this is where authors change that.
+
+*Slots:*
+- `default`  
+Custom generated rows can be placed here
+
+- `header`
+Items added here are added to the header region of the component.
+
+*Functions:*
+- `generateColumns(object): DataGridColumn`   
+Static function that creates a basic set of columns from an object representing a row.  
+
+*Events*
+- none
+
+*parts:*
+- `headerSlot`
+
+- `rowsSlot`
+
+
+**Data grid header**
+- `fast-data-grid-header`
+
+*Attributes:*
+- none
+
+*properties:*
+- `columnsData`  
+An array of `DataGridColumn` objects that define what columns will be displayed in the grid.  The order of the columns determines their order in the grid.
+
+- `headerCellItemTemplate`  
+Custom [template](https://fast.design/docs/fast-element/declaring-templates) to use when generating header cells by iterating over data.  The default template uses `fast-data-grid-header-cell`, this is where authors change that.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `headerCellsSlot`
+
+
+**Data grid header cell**
+- `fast-data-grid-header-cell`
+
+*Attributes:*
+- extends HTML Element attributes
+
+*properties:*
+- `columnData`  
+The `DataGridColumn` this column header represents.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `headerCellsSlot`
+
+
+**Data grid row**
+- `fast-data-grid-row`
+
+*Attributes:*
+- `gridTemplateColumns`  
+String that gets applied to the the css gridTemplateColumns attribute for the row
+
+*properties:*
+- `rowData`  
+The objects that contains the data to be displayed in this row.
+
+- `columnsData`  
+An array of `DataGridColumn` objects that define what columns will be displayed in the grid.  The order of the columns determines their order in the grid.
+
+- `cellItemTemplate`  
+Custom [template](https://fast.design/docs/fast-element/declaring-templates) to use when generating cells by iterating over data.  The default template uses `fast-data-grid-cell`, this is where authors change that.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `cellsSlot`
+
+**Data grid cell**
+*Component name:*
+- `fast-data-grid-cell`
+
+*Attributes:*
+- `gridColumnIndex`  
+The grid column this cell appears in.
+
+*properties:*
+- `rowData`  
+The object that contains the data to be displayed in this row.
+
+- `columnData`  
+The `DataGridColumn` this cell represents.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `cellSlot`
+
+## Implementation
```

Comment:
Can you add some code examples for how a developer would use this component?

Reply:
Added some samples and explanations.